radicle-node/src/service/tracking.rs

```diff
@@ -14,43 +14,73 @@ pub use store::Error;
 #[derive(Debug)]
 pub struct Config {
     /// Default policy, if a policy for a specific node or repository was not found.
-    default: Policy,
+    policy: Policy,
+    /// Default scope, if a scope for a specific repository was not found.
+    scope: Scope,
     /// Underlying configuration store.
     store: store::Config,
 }
 
 impl Config {
     /// Create a new tracking configuration.
-    pub fn new(default: Policy, store: store::Config) -> Self {
-        Self { default, store }
+    pub fn new(policy: Policy, scope: Scope, store: store::Config) -> Self {
+        Self {
+            policy,
+            scope,
+            store,
+        }
+    }
+
+    /// Determine whether the `id` should be fetched.
+    ///
+    /// If an entry for this `id` exists and its policy is
+    /// [`Policy::Track`], then this will return `true`.
+    ///
+    /// Otherwise, if the `id` does not exist, and the default
+    /// [`Self::scope`] is [`Scope::All`] it returns `true`.
+    ///
+    /// It returns `false` if neither of these is the case.
+    pub fn should_fetch(&self, id: &Id) -> Result<bool, Error> {
+        Ok(match self.store.repo_entry(id)? {
+            Some(entry) => entry.policy == Policy::Track,
+            None => matches!(self.scope, Scope::All),
+        })
```

Comment:
Not sure this makes sense, the scope is related to the nodes.. Instead of this function you simply want the implementation of `is_repo_tracked` without the changes in this diff.